src/modules/web3/model/weth-events.ts

```diff
@@ -0,0 +1,37 @@
+import { BigNumber, Event } from "ethers";
+
+export interface WethDepositEvent extends Event {
+  args: [string, BigNumber] & {
+    dst: string;
+    wad: BigNumber;
+  };
+}
+
+export interface WethDepositEventOptimism extends Event {
+  args: [string, BigNumber] & {
+    dst: string;
+    wad: BigNumber;
+  };
+}
+
+export interface WethDepositEventLinea extends Event {
+  args: [string, BigNumber] & {
+    dst: string;
+    wad: BigNumber;
+  };
+}
+
+export interface WethDepositEventBase extends Event {
+  args: [string, BigNumber] & {
+    dst: string;
+    wad: BigNumber;
+  };
+}
+
+export interface WethTransfetEventArbitrum extends Event {
+  args: [string, string, BigNumber] & {
+    from: string;
+    to: string;
+    value: BigNumber;
+  };
+}
```

Comment:
Any reason why we need duplicated events for different chains? AFAICT only on Arbitrum the respective event is different. But can't we just reuse the `Deposit` event signature for every other chain? This would also decrease the number of `else if` cases in the code